input/postgres/roles.go

```diff
@@ -27,15 +28,17 @@ SELECT oid,
 	FROM pg_roles r
 	 `
 
-func GetRoles(ctx context.Context, logger *util.Logger, db *sql.DB, postgresVersion state.PostgresVersion) ([]state.PostgresRole, error) {
-	stmt, err := db.PrepareContext(ctx, QueryMarkerSQL+rolesSQL)
-	if err != nil {
-		return nil, err
+func getRoles(ctx context.Context, db *sql.DB, systemType string) ([]state.PostgresRole, error) {
+	cloudSuperuserName := ""
+	if systemType == "amazon_rds" {
+		cloudSuperuserName = "rds_superuser"
+	} else if systemType == "azure_database" {
+		cloudSuperuserName = "azure_pg_admin"
+	} else if systemType == "google_cloudsql" {
+		cloudSuperuserName = "cloudsqlsuperuser"
```

Comment:
So this works, but I wonder if we get in a situation where `systemType` is not detected properly, if that could cascade to detecting this improperly. I was thinking an alternative would be to just hard-code these all (since an unintentional name clash on a different system is vanishingly unlikely), but I think in practice it won't make much difference either way, so maybe it's best to just leave it.

Reply:
Yeah, I was wondering about that too. I think I'll leave as is for now mainly because that's what the existing logic does - but not opposed to changing this in a future PR.